app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -67,24 +68,31 @@ class SparkRestClient(sparkConf: SparkConf) {
 
   private val apiTarget: WebTarget = client.target(historyServerUri).path(API_V1_MOUNT_PATH)
 
-  def fetchData(appId: String)(implicit ec: ExecutionContext): Future[SparkRestDerivedData] = {
+  def fetchData(appId: String, fetchLogs: Boolean = false)(
+    implicit ec: ExecutionContext
+  ): Future[SparkRestDerivedData] = {
     val appTarget = apiTarget.path(s"applications/${appId}")
     logger.info(s"calling REST API at ${appTarget.getUri}")
 
     val applicationInfo = getApplicationInfo(appTarget)
 
     // Limit scope of async.
+    val lastAttemptId = applicationInfo.attempts.maxBy {_.startTime}.attemptId
```

Comment:
That's right. Then I think you should keep them outside. Just add a comment there that why are they outside of the async block.